packages/phone-number-privacy/signer/test/end-to-end/get-blinded-salt.test.ts

```diff
@@ -22,88 +23,182 @@ const BLINDING_FACTOR = new Buffer('0IsBvRfkBrkKCIW6HV0/T1zrzjQSe8wRyU3PKojCnww=
-      const response = await postToSignMessage('', PRIVATE_KEY1, ACCOUNT_ADDRESS1)
+      const response = await postToSignGetSaltMessage(
+        '',
+        PRIVATE_KEY1,
+        ACCOUNT_ADDRESS1,
+        Date.now()
+      )
       expect(response.status).toBe(400)
     })
   })
 
   it('Address salt querying out of quota', async () => {
-    const response = await postToSignMessage(BLINDED_PHONE_NUMBER, PRIVATE_KEY1, ACCOUNT_ADDRESS1)
+    const response = await postToSignGetSaltMessage(
+      BLINDED_PHONE_NUMBER,
+      PRIVATE_KEY1,
+      ACCOUNT_ADDRESS1,
+      Date.now()
+    )
     expect(response.status).toBe(403)
   })
 
-  xit('Address salt querying succeeds with funded account', async () => {
-    // TODO: Figure out a common way to prefund account to provide quota
-    const response = await postToSignMessage(BLINDED_PHONE_NUMBER, PRIVATE_KEY2, ACCOUNT_ADDRESS2)
-    expect(response.status).toBe(200)
+  describe('With enough quota', () => {
+    let initialQueryCount: number
+    let timestamp: number
+    beforeAll(async () => {
+      initialQueryCount = await getQuota(PRIVATE_KEY2, ACCOUNT_ADDRESS2, IDENTIFIER)
+      timestamp = Date.now()
+    })
+
+    it('Address salt querying succeeds with unused request', async () => {
+      await replenishQuota(ACCOUNT_ADDRESS2, PRIVATE_KEY2, DEFAULT_FORNO_URL)
+      const response = await postToSignGetSaltMessage(
+        BLINDED_PHONE_NUMBER,
+        PRIVATE_KEY2,
+        ACCOUNT_ADDRESS2,
+        timestamp
+      )
+      expect(response.status).toBe(200)
+    })
+
+    it('Address salt querying with unused request increments query count', async () => {
+      const queryCount = await getQuota(PRIVATE_KEY2, ACCOUNT_ADDRESS2, IDENTIFIER)
+      expect(queryCount).toEqual(initialQueryCount + 1)
+    })
+
+    it('Address salt querying succeeds with used request', async () => {
+      await replenishQuota(ACCOUNT_ADDRESS2, PRIVATE_KEY2, DEFAULT_FORNO_URL)
+      const response = await postToSignGetSaltMessage(
+        BLINDED_PHONE_NUMBER,
+        PRIVATE_KEY2,
+        ACCOUNT_ADDRESS2,
+        timestamp
+      )
+      expect(response.status).toBe(200)
+    })
+
+    it('Address salt querying with used request does not increment query count', async () => {
+      const queryCount = await getQuota(PRIVATE_KEY2, ACCOUNT_ADDRESS2, IDENTIFIER)
+      expect(queryCount).toEqual(initialQueryCount + 1)
+    })
   })
 })
 
-async function postToSignMessage(
+async function getQuota(
+  privateKey: string,
+  account: string,
+  hashedPhoneNumber?: string,
+  authHeader?: string
+): Promise<number> {
+  const body = JSON.stringify({
+    account,
+    hashedPhoneNumber,
+  })
+  if (!authHeader) {
+    const web3 = new Web3(new Web3.providers.HttpProvider(DEFAULT_FORNO_URL))
+    const contractKit = newKitFromWeb3(web3)
+    contractKit.addAccount(privateKey)
+    authHeader = await contractKit.web3.eth.sign(body, account)
```

Comment:
I think you could just make a Web3 instance instead of instantiating and connecting contract kit, seems like overkill for to do this sign call.